src/Servers/Kestrel/test/InMemory.FunctionalTests/Http3/Http3StreamTests.cs

```diff
@@ -2489,8 +2489,8 @@ public async Task MaxRequestBodySize_ContentLengthOver_413()
 
             await requestStream.OnStreamCompletedTask.DefaultTimeout();
 
-            // TODO(JamesNK): Check for abort of request side of stream after https://github.com/dotnet/aspnetcore/issues/31970
             Assert.Contains(LogMessages, m => m.Message.Contains("the application completed without reading the entire request body."));
+            Assert.Equal("The application completed without reading the entire request body.", requestStream.AbortReadException.Message);
```

Comment:
Does this effectively get logged twice now? I'm okay with it if it's different categories and they're both debug level. Just checking.

Reply:
It's logged once at the HTTP layer and once and the transport layer.